src/util/CompressorStream.h

```diff
@@ -15,12 +15,19 @@ namespace ad_utility::streams {
 namespace io = boost::iostreams;
 using ad_utility::content_encoding::CompressionMethod;
 
+/**
+ * Takes a generator that generates strings, applies the provided
+ * compressionMethod on the fly and yields values once there is something
+ * to yield.
+ */

```

Comment:
Takes a view of strings. Behavior: The concatenation of all yielded strings is the compression, specified by the `compressionMethod` applied to the concatenation of all the strings from the `view`.